templates/project/Cargo.toml

```diff
@@ -18,26 +19,27 @@ git = "https://github.com/viamrobotics/micro-rdk.git"
 version = "0.2.9"
 rev = "a37724a"
 features = [
-  "esp32", 
+  "esp32",
   "binstart",
   {% if camera == "true" %}"camera",{% endif %}
 ]
 
 [dependencies]
 embedded-svc = "0.27"
-embedded-hal = { version = "0.2.7", features = ["unproven"]}
+embedded-hal = { version = "=0.2", features = ["unproven"]}
 log = "0.4"
 async-channel = "2"
-futures-lite = "1"
+futures-lite = "2"
 
 [build-dependencies]
 cargo_metadata = "0.18"
-embuild = "0.29"
-regex = "1.10"
-serde = { version = "1.0.156", features = ["derive"] }
-serde_json = "1.0.97"
+embuild = "0.32"
+regex = "1.11"
+serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0"
```

Comment:
since this deals with serialization stuff, should we pin the serde crates to the latest patch version? or are we relying on Cargo.lock when it comes to this?

Reply:
Relying on `Cargo.lock`. This isn't a 'pin' really, to my mind, because there is still a ^ dependency, implicitly, so would allow any semver compatible upgrade (e.g. could go to 1.1 but not 2.0).